src/openforms/js/components/admin/forms/VariableSelection.js

```diff
@@ -24,26 +25,51 @@ const VariableSelection = ({
   });
 
   const allFormVariables = (includeStaticVariables ? staticVariables : []).concat(formVariables);
+
+  const getVariableSource = variable => {
+    if (variable.source === VARIABLE_SOURCES.userDefined) {
+      return VARIABLE_SOURCES_GROUP_LABELS.userDefined;
+    }
+    if (variable.source === VARIABLE_SOURCES.component) {
+      return VARIABLE_SOURCES_GROUP_LABELS.component;
+    }
+    return VARIABLE_SOURCES_GROUP_LABELS.static;
+  };
+
   const choices = allFormVariables
```

Comment:
is there a way to order these:

1. component variables
2. user defined variables
3. static variables

I find the user defined first not too intuitive after all - and they're more likely to be used by power-users

Reply:
In the `openforms/js/components/admin/form_design/variables/constants.js` i've added a list called `VARIABLE_SOURCES_GROUP_ORDER`. This list defines the sorting order for the variable select.

Right now its set to:
1. component variables
2. user defined variables
3. static variables